api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -40,6 +40,10 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
       return true;
     }
 
+    if (request.getRequestURI().endsWith("is-username-taken")) {
```

Comment:
So rather than this, I'd prefer to add an "@Unauthenticated" annotation to select controller methods, and add logic here which checks for the presence of that annoation. I believe you can do that as indicated in the answer here:

https://stackoverflow.com/questions/29366627/spring-handlerinterceptor-how-to-access-class-annotations